app/actions/action_dropNewPin.js

```diff
@@ -40,18 +38,11 @@ export default function getLocationToSave(location) {
         coords.latitude = location.latitude;
         coords.title = 'New Pin Location';
         userData.push(coords);
-              //this will add id prop w unique id generated by db to each added pin
         userData.on("value", function(snap) {
```

Comment:
It appears Firebase has a [`.key()` method](https://www.firebase.com/docs/web/guide/saving-data.html#section-push) for the returned reference (see "Getting the Unique ID Generated by push()").

```JS
let newPinObj = userData.push();
let id = userData.key();
```